devito/types/sparse.py

```diff
@@ -607,11 +607,11 @@ def _interpolation_indices(self, variables, offset=0):
                 ub = sympy.And(p <= d.symbolic_max + self._radius, evaluate=False)
                 condition = sympy.And(lb, ub, evaluate=False)
                 mapper[d] = ConditionalDimension(p.name, self._sparse_dim,
+                                                 spacing=d.spacing,
                                                  condition=condition, indirect=True)
 
             # Track Indexed substitutions
-            idx_subs.append(OrderedDict([(v, v.subs(mapper)) for v in variables
-                                         if v.function is not self]))
+            idx_subs.append(mapper)
```

Comment:
lolz, really? what the hell did I do when I wrote it?

Reply:
Yeah the dict was `{v(x,y,z) : v(idx1, idx2)}` which isn't great for expression, this is pn;y the indices dict that way.

Reply:
```
this is pn;y the indices dict that way.
```
ehm...sorry? :D